src/main/groovy/io/seqera/wave/service/builder/ContainerBuildServiceImpl.groovy

```diff
@@ -173,7 +211,7 @@ class ContainerBuildServiceImpl implements ContainerBuildService {
             catch (FileAlreadyExistsException e) { /* ignore it */ }
             // save the dockerfile
             final containerFile = req.workDir.resolve('Containerfile')
-            Files.write(containerFile, containerFile0(req, context, spackConfig).bytes, CREATE, WRITE, TRUNCATE_EXISTING)
+            Files.write(containerFile, addLabels(containerFile0(req, context, spackConfig), req).bytes, CREATE, WRITE, TRUNCATE_EXISTING)
```

Comment:
As per suggestion above for the `addLabels` signature, here the second passed argument changes from `req` to `req.containerConfig.labels`